cluster-autoscaler/cloudprovider/gce/templates.go

```diff
@@ -227,29 +227,27 @@ func (t *templateBuilder) buildNodeFromAutoprovisioningSpec(mig *Mig, cpu int64,
 	}
 	node.Labels = labels
 
-	node.Spec.Taints = mig.spec.taints
+	node.Spec.Taints = mig.Spec().taints
 
 	// Ready status
 	node.Status.Conditions = cloudprovider.BuildReadyConditions()
 	return &node, nil
 }
 
-func buildLabelsForAutoprovisionedMig(mig *Mig, nodeName string) (map[string]string, error) {
+func buildLabelsForAutoprovisionedMig(mig Mig, nodeName string) (map[string]string, error) {
 	// GenericLabels
-	labels, err := buildGenericLabels(mig.GceRef, mig.spec.machineType, nodeName)
+	labels, err := buildGenericLabels(mig.GceRef(), mig.Spec().machineType, nodeName)
 	if err != nil {
 		return nil, err
 	}
-	if mig.spec.labels != nil {
-		for k, v := range mig.spec.labels {
-			if existingValue, found := labels[k]; found {
-				if v != existingValue {
-					return map[string]string{}, fmt.Errorf("conflict in labels requested: %s=%s  present: %s=%s",
-						k, v, k, existingValue)
-				}
-			} else {
-				labels[k] = v
+	for k, v := range mig.Spec().labels {
```

Comment:
This check would have never saved us anyway, given that earlier in this function there was:

```labels, err := buildGenericLabels(mig.GceRef, mig.spec.machineType, nodeName)```

I. e., we would have failed with nil pointer there before ever reaching the if in question.

I can add checking if mig.Spec() != nil at the beginning of every function that uses it,  if that makes it better.

Reply:
It does. Though, depending on the number of such functions I may prefer it in a separate PR. Since this PR doesn't make it any worse, I guess it's not blocking it anyway.